android/app/src/main/java/org/uwaterloo/subletr/pages/chat/ChatListingPageView.kt

```diff
@@ -0,0 +1,205 @@
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.imePadding
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import org.uwaterloo.subletr.theme.textOnSubletrPink
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rxjava3.subscribeAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
+import org.uwaterloo.subletr.R
+import org.uwaterloo.subletr.pages.account.AccountPageUiState
+import org.uwaterloo.subletr.pages.home.list.HomeListUiState
+import org.uwaterloo.subletr.services.NavigationService
+import org.uwaterloo.subletr.theme.SubletrTheme
+import org.uwaterloo.subletr.theme.primaryTextColor
+import org.uwaterloo.subletr.theme.secondaryButtonBackgroundColor
+import org.uwaterloo.subletr.theme.subletrPink
+
+@Composable
+fun ChatListingPageView(
+	modifier: Modifier = Modifier,
+	viewModel : ChatListingPageViewModel = hiltViewModel(),
+	uiState: ChatListingPageUiState = viewModel.uiStateStream.subscribeAsState(
+		ChatListingPageUiState.Loading
+	).value
+){
+	if (uiState is ChatListingPageUiState.Loading) {
+		Column(
+			modifier = modifier
+				.fillMaxSize(1.0f)
+				.imePadding(),
+			horizontalAlignment = Alignment.CenterHorizontally,
+			verticalArrangement = Arrangement.Center,
+		) {
+			CircularProgressIndicator()
+		}
+	} else if (uiState is ChatListingPageUiState.Loaded) {
+		LazyColumn(modifier = modifier) {
+			item {
+				Box(
+					modifier = Modifier
+						.fillMaxWidth()
+						.height(dimensionResource(id = R.dimen.xxl))
+						.padding(horizontal = dimensionResource(id = R.dimen.s))
+						.background(Color.Transparent),
+					contentAlignment = Alignment.BottomStart,
+				) {
+					Text(
+						stringResource(id = R.string.messages),
+						style = MaterialTheme.typography.titleMedium,
+					)
+				}
+			}
+			item {
+				DisplayEntries(uiState.contacts)
+			}
+		}
+		if (uiState.contacts.isEmpty()) {
```

Comment:
No need to render the Lazy Column if contact is empty.
It just wastes precious computing power.

Reply:
Wait why are we even rendering anything on isEmpty()? This might make sense in the context of an else if, but if it's empty we just add an extra column?

Reply:
The way it should work is that a simple `Column` without all the `LazyColumn` baggage should be rendered with the "no messages" text showing up.
So we should really have a `if(uiState.contacts.isEmpty()) {...} else {...}`